win32ss/user/ntuser/caret.c

```diff
@@ -224,8 +224,10 @@ co_IntSetCaretPos(int X, int Y)
             ThreadQueue->CaretInfo.Showing = 1;
             co_IntDrawCaret(pWnd, &ThreadQueue->CaretInfo);
          }
-
-         IntSetTimer(pWnd, IDCARETTIMER, gpsi->dtCaretBlink, CaretSystemTimerProc, TMRF_SYSTEM);
+         if (MIN_CARETBLINKRATE <= gpsi->dtCaretBlink && gpsi->dtCaretBlink < MAX_CARETBLINKRATE)
```

Comment:
Can you check, on *Windows* (Xp/2003/...), by doing a direct `SetCaretBlinkTime` call (NOTA: It would be of interest to have an apitest for this function as well, together with one for `GetCaretBlinkTime`), whether you can successfully provide smaller or larger values to the caret blink value than those `MIN/MAX_CARETBLINKRATE` ones, and whether the caret will blink in accordance?
If so, those checks here and below (and in our `IntSetCaretBlinkTime` helper function) would be definitively wrong.